src/main/java/com/carrotcreative/recyclercore/widget/ProgressRecyclerViewLayout.java

```diff
@@ -117,4 +124,131 @@ public void stopScroll()
     {
         mRecyclerView.stopScroll();
     }
+
+    // ========================================== //
+    // =============  Error State =============== //
+    // ========================================== //
+
+    public boolean isShowingErrorState()
+    {
+        if(mErrorStateView != null && mErrorStateView.getVisibility() == View.VISIBLE)
+        {
+            return true;
+        }
+        return false;
+    }
+
+    public void setErrorStateEnabled(boolean show)
+    {
+        if(mErrorStateView == null)
+        {
+            throw new IllegalStateException("Trying to setErrorStateEnabled without setting the error state View.");
+        }
+
+        if(show)
+        {
+            if(! isShowingErrorState())
+            {
+                saveCurrentViewState();
+                mErrorStateView.setVisibility(VISIBLE);
+                mRecyclerView.setVisibility(GONE);
+                mProgressBar.setVisibility(GONE);
+            }
+        }
+        else
+        {
+            restorePreviousViewState();
+        }
+    }
+
+    public void setErrorStateEnabled(View errorView, boolean show)
```

Comment:
I don't think I'm entirely sold on this interface.

I would almost rather there be a completely separate method called `setErrorStateView(View errorView)` which is only responsible for setting the error state view (Usually would be done in onCreate), and only have `setErrorStateEnabled(boolean show)` as the toggle method.